superset/connectors/sqla/models.py

```diff
@@ -391,6 +394,38 @@ def lookup_obj(lookup_metric: SqlMetric) -> SqlMetric:
 
         return import_datasource.import_simple_obj(db.session, i_metric, lookup_obj)
 
+    @property
+    def is_certified(self) -> bool:
+        try:
+            extra_object = json.loads(self.extra)
+            return bool(extra_object.get("certification", False))
+        except (TypeError, json.JSONDecodeError):
+            return False
+
+    @property
+    def certified_by(self) -> Optional[str]:
+        try:
+            extra_object = json.loads(self.extra)
+            return extra_object.get("certification", {}).get("certified_by", None)
+        except (TypeError, json.JSONDecodeError):
+            return None
+
+    @property
+    def certification_details(self) -> Optional[str]:
+        try:
+            extra_object = json.loads(self.extra)
+            return extra_object.get("certification", {}).get("details", None)
+        except (TypeError, json.JSONDecodeError):
+            return None
+
+    @property
+    def data(self) -> Dict[str, Any]:
+        attrs = ("is_certified", "certified_by", "certification_details")
+        attr_dict = {s: getattr(self, s) for s in attrs}
```

Comment:
It's a little weird that we have these related attributes floating around as separate attributes in the `attr_dict`. Does `data` need to be a flat dict, or can we keep these all in one "certification" object, sorta like how they're stored in `extra`? So, `.data` could return
```python
{
  "certification": {
    "is_certified": ...,
    "certified_by": ...,
    "certification_details":  ...
  },
  ...
}
```

Reply:
hmm, that's a good point. Unfortunately, the way the CRUD view works on the frontend is that these fields are expected to be in a flat dict here. It's possible I could modify it though, I'll take a look

Reply:
There's a decent bit of magic in the CRUD JS code, so I think keeping as a flat dict is the easiest for now. I'm going to leave this as is